sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1011,6 +1016,146 @@ If a node containing a dynamic parameter is updated through the whole graph
+recording, particularly when recording an in-order queue to specify dependencies
+as no SYCL event is returned.
+
+[source,c++]
+----
+void* Ptr = nullptr;
+size_t AllocSize = 1024;
+queue Queue {syclContext, syclDevice, {property::queue::in_order{}}};
+
+Graph.begin_recording(Queue);
+Ptr = async_malloc(Queue, usm::alloc::device, AllocSize);
+async_free(Queue, Ptr);
+----
+
+===== Supported Features [[allocation-supported-features]]
+
+Currently only device allocations are supported in graphs. Attempting to
+add allocations of any other type to a graph will result in synchronous errors
+being thrown.
+
+===== Restrictions
+
+The following restrictions apply to any graph using these allocations:
+
+* The graph cannot be finalized more than once.
+* The graph cannot be used as a sub-graph in another graph.
+* Graph memory allocation nodes cannot be updated, and graphs containing these
+nodes cannot be updated via <<whole-graph-update, Whole Graph Update>>.
+
+===== Behaviour 
+
+The semantics of `async_malloc` and `async_free` within a graph differ from the
+eager SYCL usage described in the
+link:../experimental/sycl_ext_oneapi_async_memory_alloc.asciidoc[sycl_ext_oneapi_async_memory_alloc]
+extension.
+
+* Graph memory allocations are not made directly from any default or
+user-provided Memory Pool. Each graph containing alloc/free nodes maintains its
+own pool of memory from which allocations are made. The following properties of
+a default or user-provided memory pool provided in calls to
+`async_<malloc/malloc_from_pool>` will be respected for the associated graph
+allocations, all other properties will be ignored:
+
+** The allocation type specified when creating the pool with
+`usm::alloc::<host/device/shared>`, subject to the limitations in the
+<<allocation-supported-features, supported features>> section.
+
+** `property::memory_pool::read_only` - Guarantee from the user that memory is
+only being read from, the implementation may be able to optimize in this case.
+
+** `property::memory_pool::zero_init` - Physical memory will be zero-initialized
+on first allocation during the call to `command_graph::finalize()`. It will not
+be zero-initialized again during execution of the graph. If that is required by
+the application it is the responsibility of the user to add the appropriate
+commands to the graph to do this.
+
+* `node_type::async_malloc` nodes within a graph will reserve a virtual address
+(VA) range on the device and return a pointer to the start of that VA range.
+This pointer can then be used in other graph nodes until an `async_free` node is
+added for that pointer. Returned pointers are not guaranteed to be unique. An
+implementation may return the same pointer as a previous `async_malloc` node if
+that pointer was previously freed via `async_free` at that point in the graph.
+
+* `node_type::async_free` nodes within a graph indicate that a given pointer/VA
```

Comment:
Similar to above, I think we should leave how memory is freed as an implementation detail.